usermatch/backend.js

```diff
@@ -0,0 +1,166 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const bodyParser = require('body-parser');
+
+const app = express();
+const PORT = 3000;
+
+// MongoDB Connection
+mongoose.connect('mongodb://localhost/travelplans', { useNewUrlParser: true, useUnifiedTopology: true });
+const db = mongoose.connection;
+db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+db.once('open', () => console.log('Connected to MongoDB'));
+
+// MongoDB Schema
+const userSchema = new mongoose.Schema({
+  username: String,
+  password: String,
+  name: String,
+  age: Number,
+  interests: [String],
+  travel_spots: [String],
+  hobbies: [String],
+  working_out: Boolean,
+  location: {
+    type: {
+      type: String,
+      enum: ['Point'],
+      default: 'Point',
+    },
+    coordinates: {
+      type: [Number],
+      default: [0, 0],
+    },
+  },
+});
+
+userSchema.index({ location: '2dsphere' });
+
+const User = mongoose.model('User', userSchema);
+
+// Middleware
+app.use(bodyParser.json());
+
+// Endpoints
+app.post('/login', async (req, res) => {
+  const { username, password } = req.body;
+  try {
+    const user = await User.findOne({ username, password });
+    if (user) {
+      return res.status(200).json({ message: 'Login successful' });
+    } else {
+      return res.status(401).json({ error: 'Invalid credentials' });
+    }
+  } catch (error) {
+    return res.status(500).json({ error: error.message });
+  }
+});
+
+app.post('/userprofile', async (req, res) => {
+  const { username, password, name, age, interests, travel_spots, hobbies, working_out, latitude, longitude } = req.body;
+
+  try {
+    let user = await User.findOne({ username });
+    if (user) {
```

Comment:
For now, let's do an equality check between the given password and the password stored in the DB.
req.body.password === user.password.
A random person should not be able to update an existing user profile by guessing the username randomly.
Longer term, we will change this to improve security, since we should be wary of storing password as-is, without encryption